src/section/shared_state.rs

```diff
@@ -60,11 +62,19 @@ impl SharedState {
             );
         }
 
-        if self.our_history.len() > 1 && *self != new {
+        if self.our_history.len() > 1 {
+            if *self != new {
```

Comment:
I really want to get this log_or_panic gone ASAP. In this case here will it matter? I mean, it's an Error and a valid Error, why log or panic, if we want to test this we can test for that error. 
If these are semilattice then perhaps we can be even more smart?

Reply:
was just following the previous implementation.
I think the idea here is that shall not be happen, and if happens during test, we want it to be exposed immediately and explicitly.

However, I think there is missing `return` for these `error branches` ? @madadam

Reply:
Yes, I understand but let's get to launch now and no more log or panic. We don't panic unless we have an unrecoverable error. I am not sure we have that here so we should return the error or fix the error. If we return it we should log Error and if we fix it then perhaps log Debug? 
Panic in tests would mean we did neither and the production code has to do one of these things (return or fix). I say now is the time to stop using log_or_panic and getting rid of it when we are working on those bits of code. 

It's a better cleaner solution to not use this from this point forward.  :+1:

Reply:
ok, I will change it to log and return with an error

Reply:
Or just `?` if it can be handled upstream.